server/public/plugin/client_rpc.go

```
@@ -451,22 +475,70 @@ type Z_ServeHTTPArgs struct {
-		connection, err := g.muxBroker.Accept(serveHTTPStreamId)
-		if err != nil {
-			g.log.Error("Plugin failed to ServeHTTP, muxBroker couldn't accept connection", mlog.Uint("serve_http_stream_id", serveHTTPStreamId), mlog.Err(err))
+		connection, err := g.muxBroker.Accept(responseStreamID)
+		if err != nil || !responseStream.Set(connection) {
+			if err != nil {
+				g.log.Error("Plugin failed to ServeHTTP, muxBroker couldn't accept connection", mlog.Uint("serve_http_stream_id", responseStreamID), mlog.Err(err))
+			}
 			return
 		}
-		defer connection.Close()
+		defer responseStream.Close()
```

golangci-lint (2.13.2)

[error] Error return value of `responseStream.Close` is not checked

(errcheck)